webui_streamlit.py

```diff
@@ -1263,72 +1289,85 @@ def sample(init_data, x, conditioning, unconditional_conditioning, sampler_name)
-			        jpg_sample=save_as_jpg,
-			        job_info=job_info
+			        jpg_sample=save_as_jpg
 			    )
 
 			if initial_seed is None:
 				initial_seed = seed
 
 			init_img = output_images[0]
+			
+			if do_color_correction and correction_target is not None:
+				init_img = Image.fromarray(cv2.cvtColor(exposure.match_histograms(
+					cv2.cvtColor(
+					    np.asarray(init_img),
+					cv2.COLOR_RGB2LAB
+					),
+				    correction_target,
+				    channel_axis=2
+				    ), cv2.COLOR_LAB2RGB).astype("uint8"))
+				
 			if not random_seed_loopback:
 				seed = seed + 1
 			else:
 				seed = seed_to_int(None)
+				
 			denoising_strength = max(denoising_strength * 0.95, 0.1)
 			history.append(init_img)
 
-		if not skip_grid:
+		if save_grid:
+			full_path = os.path.join(os.getcwd(), outpath)
+			
 			grid_count = get_next_sequence_number(outpath, 'grid-')
-			grid = image_grid(history, batch_size, force_n_rows=1)
-			grid_file = f"grid-{grid_count:05}-{seed}_{prompt.replace(' ', '_').translate({ord(x): '' for x in invalid_filename_chars})[:128]}.{grid_ext}"
+			grid = image_grid(history, batch_size)
+			#grid_file = f"grid-{grid_count:05}-{seed}_{prompt.replace(' ', '_').translate({ord(x): '' for x in invalid_filename_chars})[:220-len(full_path)]}.{grid_ext}"
+			grid_file = f"grid-{grid_count:05}-{seed}_{slugify(prompt.replace(' ', '_'))[:220-len(full_path)]}.{grid_ext}"
 			grid.save(os.path.join(outpath, grid_file), grid_format, quality=grid_quality, lossless=grid_lossless, optimize=True)
 
-
 		output_images = history
 		seed = initial_seed
 
 	else:
 		output_images, seed, info, stats = process_images(
-	        outpath=outpath,
-	    func_init=init,
-	    func_sample=sample,
-	    prompt=prompt,
-	    seed=seed,
-	    sampler_name=sampler_name,
-	    skip_save=False,
-	    skip_grid=False,
-	    batch_size=batch_size,
-	    n_iter=n_iter,
-	    steps=ddim_steps,
-	    cfg_scale=cfg_scale,
-	    width=width,
-	    height=height,
-	    prompt_matrix=False,
-	    use_GFPGAN=use_GFPGAN,
-	    use_RealESRGAN=use_RealESRGAN,
-	    realesrgan_model_name="",
-	    fp=fp,
-	    normalize_prompt_weights=normalize_prompt_weights,
-	    init_img=init_img,
-	    init_mask=init_mask,
-	    keep_mask=keep_mask,
-	    mask_blur_strength=2,
-	    denoising_strength=denoising_strength,
-	    resize_mode=0,
-	    uses_loopback=loopback,
-	    sort_samples=False,
-	    write_info_files=write_info_files,
-	    jpg_sample=True
-	)
+		        outpath=outpath,
+		        func_init=init,
+		        func_sample=sample,
+		        prompt=prompt,
+		        seed=seed,
+		        sampler_name=sampler_name,
+		        skip_save=not save_individual_images,
+		        skip_grid=not save_grid,
+		        batch_size=batch_size,
+		        n_iter=n_iter,
+		        steps=ddim_steps,
+		        cfg_scale=cfg_scale,
+		        width=width,
+		        height=height,
+		        prompt_matrix=prompt_matrix,
```

Comment:
```suggestion
		        prompt_matrix=separate_prompts,
```
Did you perhaps forget to change this one?